lib/rspec_profiling/collectors/statsd.rb

```diff
@@ -84,14 +85,21 @@ def formatDesc(description)
         ::Digest::SHA1.hexdigest(description)[8..15]
       end
 
-      def formatFile(path) 
-        path.gsub('/', '.').gsub(/\.rb$|^\.[^.]*\./, '')
+      def formatFile(path, max = 0) 
+        str = path.gsub('/', '.').gsub(/\.rb$|^\.[^.]*\./, '')
+        if max > 0 then 
```

Comment:
This entire block can be replaced by a more idiomatic and shorter statement:
`str = (str.split('.')[0..max-1] << parts[-1]).join('.') if max > 0`

Reply:
Also, `format_file` instead of `formatFile`

Reply:
That is a very dense statement, which can be good.  However I find it really odd to have the condition at the end of the statement, particularly with a longer statement like this.  Is there any advantage to this or is it purely style of syntax?

Reply:
Pretty much purely style of syntax. I do agree with you from the perspective of someone new to Ruby it's confusing, but you get used to it ;)

As far as the single line or multiple line version, to me it's easier to understand, because the entire manipulation of the string reads left-to-right on one line without having to figure out if those intermediary variables that you assign get used elsewhere.

That said, I have a bug in my statement hahaha since `parts` no longer exists. I leave it to you to fix it.